app/controllers/RaceController.java

```diff
@@ -0,0 +1,97 @@
+package controllers;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import models.Race;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import play.mvc.BodyParser;
+import play.mvc.Controller;
+import play.mvc.Result;
+import repository.interfaces.RaceRepository;
+
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Function;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static play.libs.Json.toJson;
+
+public class RaceController extends Controller {
+    private final RaceRepository raceRepository;
+
+    @Inject
+    public RaceController(RaceRepository raceRepository) {
+        this.raceRepository = raceRepository;
+    }
+
+    public CompletionStage<Result> getAllRaces() {
+        return raceRepository.getAllRaces().thenApplyAsync(races -> {
+            return ok(races);
+        }).exceptionally(ex -> {
+            Result res = null;
+            switch (ExceptionUtils.getRootCause(ex).getClass().getSimpleName()){
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this "switch" statement by "if" statements to increase readability. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1301)